File: cash.c
#include <math.h>
#include <cs50.h>
#include <stdio.h>

int change_owed(void);
int coins_count(int);

int main(void) 
{   
    int change = change_owed();
    printf("%i\n", coins_count(change));
}

int coins_count(int change)
{
    int coins[] = {25, 10, 5, 1};
    int count = 0;
    
    for (int i = 0; i < 4; i++)
    {
        while (change >= coins[i])
        {
            change -= coins[i];
            count++;
        }
    }
    return count;
}

int change_owed()
{
    float amount;
    do
    {
        amount = get_float("Change owed: ");       
    }
    while (amount <= 0);
    return (int) round(amount * 100);
}

File: cash.py
from cs50 import get_float


def main():
    change = int(change_owed())
    print(f"{coins_count(change)}")


def coins_count(change):
    # Initialize coins and count
    coins = [25, 10, 5, 1]
    count = 0

    # subtract highest possible coin from remaining change
    for coin in coins:
        while change >= coin:
            print(f"change left: {change}; coin used: {coin}")
            change -= coin
            count += 1
    return count


# Get input from user
def change_owed():
    while True:
        amount = get_float("Change owed: ")
        if amount > 0:
            break
    return amount * 100


if __name__ == "__main__":
    main()